### PR TITLE
Add support for "on delete cascade" column option

### PR DESCRIPTION
**Pull request by c7hm4r**
This introduces some breaking changes.

Is it a good idea to have a lifetime parameter in `Parser`?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,3 +11,4 @@ Cargo.lock
 
 # IDEs
```
[the remark this answers is not in the document]
Comment (c7hm4r) [reply .gitignore:12]: Ok cool. I didn’t know that technique.

File: src/ast/ddl.rs [hunk 2]
```diff
@@ -158,7 +158,13 @@ pub enum ColumnOption {
     /// <foreign_table> (<referred_columns>)`).
     ForeignKey {
         foreign_table: ObjectName,
-        referred_columns: Vec<Ident>,
+        referred_columns: Option<Vec<Ident>>,
```
[the remark this answers is not in the document]
Comment (c7hm4r) [reply src/ast/ddl.rs:161]: Ok, I will revert that.

File: src/ast/ddl.rs [hunk 3]
```diff
@@ -158,7 +158,13 @@ pub enum ColumnOption {
...
+    },
+    // ON { UPDATE | DELETE }
```
[the remark this answers is not in the document]
Comment (c7hm4r) [reply src/ast/ddl.rs:163]: Your proposal sounds more suitable, actually.

**Comment by c7hm4r** [comment]
I reworked the PR. I will create a new RP for the new dialect option.

File: src/parser.rs [hunk 4]
```diff
@@ -1017,10 +1017,31 @@ impl Parser {
             ColumnOption::Unique { is_primary: false }
         } else if self.parse_keyword("REFERENCES") {
             let foreign_table = self.parse_object_name()?;
-            let referred_columns = self.parse_parenthesized_column_list(Mandatory)?;
+            let referred_columns = self.parse_parenthesized_column_list(Optional)?;
```
Comment (nickolay) [inline src/parser.rs:1020]: Please add a comment, e.g. `// PostgreSQL allows omitting the column list to refer to the foreign table by its PK`

File: src/parser.rs [hunk 5]
```diff
@@ -1017,10 +1017,31 @@ impl Parser {
...
+            let mut on_delete = None;
+            let mut on_update = None;
+            while self.parse_keyword("ON") {
+                if self.parse_keyword("DELETE") {
+                    if on_delete == None {
+                        on_delete = Some(self.parse_reference_change_action()?);
+                    } else {
+                        return self
+                            .expected("ON DELETE option not more than once", self.peek_token());
```
Comment (nickolay) [inline src/parser.rs:1029]: I don't think this deserves a special error message, and they add some complexity. Would this work instead?
```
            let (mut on_delete, mut on_update) = (None, None);
            loop {
                if on_delete.is_none() && self.parse_keywords(vec!["ON", "DELETE"]) {
                    on_delete = Some(self.parse_reference_change_action()?);
                } else if on_update.is_none() && self.parse_keywords(vec!["ON", "UPDATE"]) {
                    on_update = Some(self.parse_reference_change_action()?);
                } else {
                    break;
                }
            }
```

File: src/parser.rs [hunk 6]
```diff
@@ -1017,10 +1017,31 @@ impl Parser {
...
+                    }
+                } else if self.parse_keyword("UPDATE") {
+                    if on_update == None {
+                        on_update = Some(self.parse_reference_change_action()?);
+                    } else {
+                        return self
+                            .expected("ON UPDATE option not more than once", self.peek_token());
+                    }
+                }
```
Comment (nickolay) [inline src/parser.rs:1038]: If the above idea doesn't work for some reason, here's an `else` branch missing.

File: src/ast/ddl.rs [hunk 7]
```diff
@@ -200,3 +215,28 @@ fn display_constraint_name<'a>(name: &'a Option<Ident>) -> impl fmt::Display + '
     }
     ConstraintName(name)
 }
+
+/// `<reference_change_action> =
```
[the remark this answers is not in the document]
Comment (c7hm4r) [reply src/ast/ddl.rs:219]: Sorry, I simply forgot to also rename `reference_change_action` to `referential_action`.

File: src/parser.rs (hunk 5 above)
Comment (c7hm4r) [reply src/parser.rs:1029]: I added some tests to support that your proposal works.

File: src/ast/ddl.rs [hunk 8]
```diff
@@ -155,10 +155,16 @@ pub enum ColumnOption {
         is_primary: bool,
     },
     /// A referential integrity constraint (`[FOREIGN KEY REFERENCES
-    /// <foreign_table> (<referred_columns>)`).
+    /// <foreign_table> (<referred_columns>)
+    /// [ON DELETE <reference_change_action>]
```
[the remark this answers is not in the document]
Comment (c7hm4r) [reply src/ast/ddl.rs]: probably this is resolved with the new proposal

File: src/ast/ddl.rs [hunk 9]
```diff
@@ -155,10 +155,16 @@ pub enum ColumnOption {
...
+    /// [ON UPDATE <reference_change_action>]`).
+    ///
+    /// The `ON DELETE` option may be placed after the `ON UPDATE` option.
```
[the remark this answers is not in the document]
Comment (c7hm4r) [reply src/ast/ddl.rs]: I expressed that formally now

**Pull request merged**